### DISCUSSION TITLE
"Add account" location

### DISCUSSION THREAD

**Discussion by IBBoard** [CodedOre/NewCaw#68] [Design]
Is there any technical reason why we can't move the "Add Account" buttons to the left-panel?

It feels a little odd to have to go under "Preferences" to manage my accounts. If it was "Settings" then it might make a bit more sense, but mostly because of conventions in other apps. But my account isn't a preference. And it feels natural to have the "switch accounts" action in with the "add another one to switch to" action.

**Comment by CodedOre** [comment]
This has to do with how the account switcher will be designed. I had some different ideas on how to designed the UI for changing views and accounts. I'll share them tomorrow. But in not all of them I would've said an "Add Account" button would fit. 

Because of this and because its something you set up once and leave as it is afterwards, I thought it fits well as an setting.

**Reply by CodedOre** [reply]
> I'll share them tomorrow.

Seems like I completely forgot about that...\
Sorry about that.

Anyway, like I mentioned before, during the time I developed different concepts for how the account and view switcher could look, so I'll just show them in chronological order.

---

First of all, the original concept. This concept was just created as a base from which the other concepts derivated, by using the design from [#IBBoard/cawbird177](https://github.com/IBBoard/cawbird/issues/177).

<img src="https://user-images.githubusercontent.com/47981497/215091053-9573e27b-1e64-40af-8827-3822fcbcb355.png" width="75%" height="75%"/>


---

Back in September 2021, I wanted to look into different ideas so I could move the account switcher out of the menu popover. The first concept lead to this design using the `Adw.Flap`:

<img src="https://user-images.githubusercontent.com/47981497/215094520-63889611-b123-4861-89a8-cd9aeb847f74.png" width="45%" height="45%"/> <img src="https://user-images.githubusercontent.com/47981497/215094524-7f2747d4-2016-449d-9a06-d548315b385e.png" width="45%" height="45%"/>

---

In that time I also had made a concept for a fullscreen switch. If I recall correctly, the motivation was to show more of the accounts so the user could more easily identify different accounts and platforms, but looking back at it it isn't the best design in my view:

<img src="https://user-images.githubusercontent.com/47981497/215095868-99d61db6-2c23-40b9-9591-676af0a8a662.png" width="45%" height="45%"/> <img src="https://user-images.githubusercontent.com/47981497/215095873-89e8f96b-5c77-44df-8967-9f2435327e5c.png" width="45%" height="45%"/>

---

After that design work, I decided to use the flap concept as the plan, which leads to the current design of `MainView`. However, more recently, as in late August of last year, I had another run at creating concepts. The motivation was as follows: Four views are not really enough to display all things we want to show. For this reason, we had the plan to place bookmarks and favorites in the list view, so they don't need their own view.

However, inspired by Gnome's control panel, I created then a design moving all the view switchers to the flap:

<img src="https://user-images.githubusercontent.com/47981497/215097088-7b2d8c7f-480c-423f-8966-1e7102c2d196.png" width="45%" height="45%"/> <img src="https://user-images.githubusercontent.com/47981497/215097095-04b543b9-90f1-43d4-bf48-4925ce770acd.png" width="45%" height="45%"/>

The account switcher would now be accessed after clicking on the row with the account.

---

While I liked the idea, I got the feedback from the Gnome designers back then that it would make accessing often used views difficult. So, this the concept, which would only place views in the sidebar which are less often used. As of now, this is probably the closest to being my currently planned design:

<img src="https://user-images.githubusercontent.com/47981497/215098175-38c91431-e0f9-40c0-9492-87805c8a969c.png" width="45%" height="45%"/> <img src="https://user-images.githubusercontent.com/47981497/215098178-b9fca32b-9668-478f-9341-65e5ce0dc353.png" width="45%" height="45%"/>

If I think now about this, the idea to hide other accounts behind clicking the currently active account might not be that visible, so in this design the other accounts could be displayed below the views remaining in the sidebar.

---

So, this pretty much sums up all my designs I had for the `MainView`. Basically, **the TL:DR is: I still have to find a good design for how to allow switch all accounts and views we might have at the end, and it will be a lot of UI already there, so adding an "Add Account" you would rarely accessed is not something I would want to do.**

Alright, that it's for now. I still have a project and an exam to finish in this and the next week, so I'll probably be quite silent in the next days as well.

**Reply by IBBoard** [reply]
Makes more sense not to include it if the list is going to be more than just a list of profiles. I was just working off what's there now and treating it like the profile switcher button in current Cawbird.

I'd like a better location than "preferences", though. My account is not a preference. It's a setting. But Gnome naming conventions are that we have app preferences and not settings.
